datahub-upgrade/src/main/java/com/linkedin/datahub/upgrade/removeunknownaspects/RemoveUnknownAspectsStep.java

```diff
@@ -0,0 +1,41 @@
+package com.linkedin.datahub.upgrade.removeunknownaspects;
+
+import com.linkedin.datahub.upgrade.UpgradeContext;
+import com.linkedin.datahub.upgrade.UpgradeStep;
+import com.linkedin.datahub.upgrade.UpgradeStepResult;
+import com.linkedin.datahub.upgrade.impl.DefaultUpgradeStepResult;
+import com.linkedin.gms.factory.telemetry.TelemetryUtils;
+import com.linkedin.metadata.entity.EntityService;
+import java.util.HashMap;
+import java.util.function.Function;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+
+
+@Slf4j
+@RequiredArgsConstructor
+public class RemoveUnknownAspectsStep implements UpgradeStep {
+
+  private static final String INVALID_CLIENT_ID_ASPECT = "clientId";
+
+  private final EntityService _entityService;
+
+  @Override
+  public String id() {
+    return this.getClass().getSimpleName();
+  }
+
+  @Override
+  public boolean skip(UpgradeContext context) {
+    return false;
+  }
+
+  @Override
+  public Function<UpgradeContext, UpgradeStepResult> executable() {
+    return upgradeContext -> {
+      _entityService.deleteAspect(TelemetryUtils.CLIENT_ID_URN, INVALID_CLIENT_ID_ASPECT,
```

Comment:
Naming: Why name this step like this if its so specific? Not sure that is useful.

Reply:
Renamed :)